frontends/krita/krita_comfy/widgets/prompt.py

```diff
@@ -1,37 +1,62 @@
-import re
-from PyQt5.QtCore import Qt
-from PyQt5.QtGui import QSyntaxHighlighter, QColor
+from PyQt5.QtCore import Qt, QRect
+from PyQt5.QtGui import QFocusEvent, QKeyEvent, QTextCursor
 from PyQt5.QtWidgets import QPlainTextEdit, QSizePolicy, QVBoxLayout
-from ..config import Config
-from ..utils import auto_complete_LoRA, auto_complete_embedding, re_lora, re_embedding
-
-
-class QPromptHighLighter(QSyntaxHighlighter):
-    def __init__(self, cfg: Config, *args, **kwargs):
-        super(QSyntaxHighlighter, self).__init__(*args, **kwargs)
-
-        self.cfg = cfg
-        self.highlighters = [
-            [re_lora, auto_complete_LoRA],
-            [re_embedding, auto_complete_embedding]]
 
-    def highlightBlock(self, line):
-        for expression, func in self.highlighters:
-            for m in re.finditer(expression, line):
-                valid, names = func(self.cfg, m.group(1))
-                color: QColor = QColor(Qt.green)
-                color = color if valid else QColor(Qt.red)
-                color = color if len(names) < 2 else QColor(Qt.yellow)
-                self.setFormat(m.start(0), m.end(0)-m.start(0), color)
+from ..config import Config
+from .prompt_complete import QPromptCompleter, QPromptHighLighter
 
 
 class QPromptEdit(QPlainTextEdit):
+    completer: QPromptCompleter = None
+
     def __init__(self, placeholder="Enter prompt...", num_lines=5, *args, **kwargs):
         super(QPromptEdit, self).__init__(*args, **kwargs)
         self.setPlaceholderText(placeholder)
         self.setFixedHeight(self.fontMetrics().lineSpacing() * num_lines)
         self.setSizePolicy(QSizePolicy.MinimumExpanding, QSizePolicy.Maximum)
 
+    def setCompleter(self, completer: QPromptCompleter):
+        if self.completer:
+            self.disconnect(self.completer)
+        if not completer:
+            return
+
+        completer.setWidget(self)
+        self.completer = completer
+        self.completer.insertText.connect(self.insertCompletion)
+
+    def insertCompletion(self, text):
+        tc = self.textCursor()
+        extra = (len(text) - len(self.completer.completionPrefix()))
+        tc.movePosition(QTextCursor.Left)
+        tc.movePosition(QTextCursor.EndOfWord)
+        tc.insertText(text[-extra:])
+        self.setTextCursor(tc)
+        self.completer.hide_popup()
```

Comment:
```suggestion
    def insertCompletion(self, text):
        tc = self.textCursor()
        # Delete the current word after a colon
        while tc.positionInBlock() > 0 and not tc.atBlockStart() and tc.selectedText() != ":":
            tc.deletePreviousChar()
            tc.movePosition(QTextCursor.Left, QTextCursor.KeepAnchor)   
        # Insert the new word
        tc.movePosition(QTextCursor.EndOfWord)
        tc.insertText(text)
        self.setTextCursor(tc)
        self.completer.hide_popup()
```

The current implementation successfully completes the embedding/lora name, however, it would be better if it replaced the current written name. Something like this allows users start writing names in lower case and getting the correct name of the resource.